api/app/Models/BaseModel.php

```diff
@@ -1,40 +1,42 @@
-<?php
-
-namespace App\Models;
+<?php namespace App\Models;
 
+use App\Exceptions\ValidationException;
+use Carbon\Carbon;
 use Bosnadev\Database\Traits\UuidTrait;
-use Illuminate\Database\Eloquent\Model;
+use Illuminate\Support\MessageBag;
+use Illuminate\Validation\Factory as Validator;
 
-abstract class BaseModel extends Model
+abstract class BaseModel extends \Spira\Repository\Model\BaseModel
 {
     use UuidTrait;
 
     public $incrementing = false;
 
     /**
-     * The accessors to append to the model's array form.
-     *
-     * @var array
+     * @var Validator
      */
-    public $appends = ['self'];
+    protected $validator;
 
     /**
-     * Get the access route for the entity.
-     *
-     * @return string
+     * @var MessageBag|null
      */
-    abstract public function entityRoute();
+    protected $errors;
 
     /**
-     * Get the user's first name.
-     *
-     * @param string $value
-     *
-     * @return string
+     * @return array
      */
-    public function getSelfAttribute()
+    public function getValidationRules()
     {
-        return url($this->entityRoute().'/'.$this->{$this->primaryKey});
+        return [];
+    }
+
+    protected function getValidator()
+    {
+        if (is_null($this->validator)) {
+            $this->validator = \App::make('validator');
```

Comment:
It's a basic Laravel/Lumen validation factory class.
Core component.
For now it's the only possible way, as it injects translation service.

We'd better leave it for now and change it for Assertions in the future. (If we will have enougth inner strength for this =))))